templates/terraform/custom_expand/dns_managed_zone_private_visibility_config.go.erb

```diff
@@ -0,0 +1,73 @@
+<%- # the license inside this block applies to this file
+	# Copyright 2020 Google Inc.
+	# Licensed under the Apache License, Version 2.0 (the "License");
+	# you may not use this file except in compliance with the License.
+	# You may obtain a copy of the License at
+	#
+	#     http://www.apache.org/licenses/LICENSE-2.0
+	#
+	# Unless required by applicable law or agreed to in writing, software
+	# distributed under the License is distributed on an "AS IS" BASIS,
+	# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	# See the License for the specific language governing permissions and
+	# limitations under the License.
+-%>
+func expand<%= prefix -%><%= titlelize_property(property) -%>(v interface{}, d TerraformResourceData, config *Config) (interface{}, error) {
+	l := v.([]interface{})
+	if len(l) == 0 || l[0] == nil {
+		// The API won't remove the the field unless an empty network array is sent.
+		transformed := make(map[string]interface{})
+		emptyNetwork := make([]interface{}, 0)
+		transformed["networks"] = emptyNetwork
+		return transformed, nil
+	}
+	raw := l[0]
+	original := raw.(map[string]interface{})
+	transformed := make(map[string]interface{})
+
+	transformedNetworks, err := expandDNSManagedZonePrivateVisibilityConfigNetworks(original["networks"], d, config)
+	if err != nil {
+		return nil, err
+	} else if val := reflect.ValueOf(transformedNetworks); val.IsValid() && !isEmptyValue(val) {
+		transformed["networks"] = transformedNetworks
+	}
+
+	return transformed, nil
+}
+
+func expand<%= prefix -%><%= titlelize_property(property) -%>Networks(v interface{}, d TerraformResourceData, config *Config) (interface{}, error) {
```

Comment:
You could probably include templating that generates these expanders: https://github.com/GoogleCloudPlatform/magic-modules/blob/master/templates/terraform/expand_property_method.erb#L186-L192

I don't feel particularly strongly about either approach. I think it's less likely the expanders drift by using the templating, but more likely we introduce a hard-to-debug difference if we ever edit the templating in expand_property_method.erb & not here.

Reply:
I slightly lean towards keeping them explicitly defined just for debugging. I thought about adding a new flag just to enable the the nested properties generation so we can reuse the templating there, but didn't think it was worth it since this seems to be the first case where it was relevant.